truss-chains/truss_chains/utils.py

```diff
@@ -131,33 +141,46 @@ def get_free_port() -> int:
         return port
 
 
+# NOTE(dynamic-chainlet-config): Predict URLs
+# for services that belong to a Chainlet are
+# auto-populated through dynamic config.
 def override_chainlet_to_service_metadata(
     chainlet_to_service: Mapping[str, definitions.ServiceDescriptor],
-):
+) -> Mapping[str, definitions.DeployedServiceDescriptor]:
+    chainlet_to_deployed_service: Dict[str, definitions.DeployedServiceDescriptor] = {}
+
     # Override predict_urls in chainlet_to_service ServiceDescriptors if dynamic_chainlet_config exists
     dynamic_chainlet_config_str = dynamic_config_resolver.get_dynamic_config_value_sync(
         definitions.DYNAMIC_CHAINLET_CONFIG_KEY
     )
-    if dynamic_chainlet_config_str:
-        dynamic_chainlet_config = json.loads(dynamic_chainlet_config_str)
-        for (
-            chainlet_name,
-            service_descriptor,
-        ) in chainlet_to_service.items():
-            if chainlet_name in dynamic_chainlet_config:
+
+    if not dynamic_chainlet_config_str:
+        raise RuntimeError(
+            f"No '{definitions.DYNAMIC_CHAINLET_CONFIG_KEY}' found. Cannot override Chainlet configs."
+        )
+
+    dynamic_chainlet_config = json.loads(dynamic_chainlet_config_str)
+
+    for (
+        chainlet_name,
+        service_descriptor,
+    ) in chainlet_to_service.items():
+        if chainlet_name not in dynamic_chainlet_config:
+            raise RuntimeError(
+                f"Chainlet '{chainlet_name}' not found in '{definitions.DYNAMIC_CHAINLET_CONFIG_KEY}'."
+            )
+
+        chainlet_to_deployed_service[chainlet_name] = (
+            definitions.DeployedServiceDescriptor(
+                name=service_descriptor.name,
+                options=service_descriptor.options,
                 # We update the predict_url to be the one pulled from the dynamic_chainlet_config
-                service_descriptor.predict_url = dynamic_chainlet_config[chainlet_name][
-                    "predict_url"
-                ]
-            else:
-                logging.debug(
-                    f"Skipped override for chainlet '{chainlet_name}': not found in {definitions.DYNAMIC_CHAINLET_CONFIG_KEY}."
-                )
-    else:
-        logging.debug(
-            f"No {definitions.DYNAMIC_CHAINLET_CONFIG_KEY} found, skipping overrides."
+                predict_url=dynamic_chainlet_config[chainlet_name]["predict_url"],
```

Comment:
Can you please remove above comment. My guideline for comments is: explain *why* something happens (but not what). The latter should always be self evident from the code, or you need better names / strucutre. The former should ideally also be self evident, but in some cases it helps to put it into words or add extra context...